### PR TITLE
fix(unfurl): use Node https.get to avoid prefetch; don't fetch non-HTTPS links

### PR DESCRIPTION
**Pull request by dchest**
On desktop, XMLHttpRequest prefetches everything mentioned in `link`
response headers. To avoid it, use Node's https.get on desktop.

On mobile, still use XHR (it shouldn't prefetch anything, hopefully).

#### Relevant info and issue/PR links 

https://app.clubhouse.io/peerio/story/14391/url-unfurl-tries-to-load-css-blocked-by-content-security-policy
 
#### Testing instructions  

Ensure linked image previews work on both desktop and mobile.

----
### Repository owner

- [ ] Was tested and can be merged.
- [ ] PR name follows conventional changelog format.

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### CODE REVIEW

File: src/helpers/unfurl.ts [hunk 1]
```diff
@@ -19,7 +19,35 @@ export function getUrls(str: string): string[] {
...
 }
-export function getContentHeaders(url: string) {
+
+export const getContentHeaders = config.isMobile ? getContentHeadersXHR : getContentHeadersNode;
+
+let https; // Node module 'https', loaded when not on mobile.
+
+// Exported for unit testing, use getContentHeaders.
+export function getContentHeadersNode(url: string): Promise<HeaderDict> {
+    if (!https) https = require('https'); // eslint-disable-line global-require
```
Comment (anri-asaturov) [inline src/helpers/unfurl.ts]: does this mean all requests are gonna be going through interop?
You think it's ok performance wise?

Comment (dchest) [reply src/helpers/unfurl.ts]: Not much interop there, unlike mobile, Node.js is embedded into the same runtime in Electron. Didn't test performance, but also didn't notice any slowdowns.

**Comment by ebilg** [comment]
was not able to build mobile (ios) with this branch
```
loading custom config for packager
Scanning folders for symlinks in /Users/ebilg/peerio-mobile/node_modules (23ms)
Scanning folders for symlinks in /Users/ebilg/peerio-mobile/node_modules (20ms)
Loading dependency graph, done.
warning: the transform cache was reset.

Unable to resolve module `https` from `/Users/ebilg/peerio-mobile/app/lib/peerio-icebear/helpers/unfurl.js`: Module `https` does not exist in the Haste module map

This might be related to https://github.com/facebook/react-native/issues/4968
To resolve try the following:
  1. Clear watchman watches: `watchman watch-del-all`.
  2. Delete the `node_modules` folder: `rm -rf node_modules && npm install`.
  3. Reset Metro Bundler cache: `rm -rf /tmp/metro-bundler-cache-*` or `npm start -- --reset-cache`.  4. Remove haste cache: `rm -rf /tmp/haste-map-react-native-packager-*`.

JS compilation failed. Please check the sources
Command PhaseScriptExecution failed with a nonzero exit code
```

**Comment by ebilg** [comment]
getting this even after trying to reinstall node modules as suggested
```
loading custom config for packager
Scanning folders for symlinks in /Users/ebilg/peerio-mobile/node_modules (23ms)
Scanning folders for symlinks in /Users/ebilg/peerio-mobile/node_modules (20ms)
Loading dependency graph, done.
warning: the transform cache was reset.

Unable to resolve module `https` from `/Users/ebilg/peerio-mobile/app/lib/peerio-icebear/helpers/unfurl.js`: Module `https` does not exist in the Haste module map

This might be related to https://github.com/facebook/react-native/issues/4968
To resolve try the following:
  1. Clear watchman watches: `watchman watch-del-all`.
  2. Delete the `node_modules` folder: `rm -rf node_modules && npm install`.
  3. Reset Metro Bundler cache: `rm -rf /tmp/metro-bundler-cache-*` or `npm start -- --reset-cache`.  4. Remove haste cache: `rm -rf /tmp/haste-map-react-native-packager-*`.

JS compilation failed. Please check the sources
Command PhaseScriptExecution failed with a nonzero exit code
```

**Comment by dchest** [comment]
Pls re-review.

**Comment by BlueMona** [comment]
Was tested and can be merged.

**Pull request merged**